smartsim/_core/launcher/slurm/slurmLauncher.py

```diff
@@ -188,14 +188,31 @@ def stop(self, step_name: str) -> StepInfo:
         stepmap = self.step_mapping[step_name]
         if stepmap.managed:
             step_id = str(stepmap.step_id)
-            # Check if step_id is part of colon-separated run
-            # if that is the case, stop parent job step because
+            # Check if step_id is part of colon-separated run,
+            # this is reflected in a '+' in the step id,
+            # so that the format becomes 12345+1.0.
+            # If we find it it can mean two things:
+            # a MPMD srun command, or a heterogeneous job.
+            # If it is a MPMD srun, then stop parent step because
             # sub-steps cannot be stopped singularly.
-            if "+" in step_id:
+            sub_step = "+" in step_id
+            het_job = os.getenv("SLURM_HET_SIZE") is not None
+            # If it is a het job, we can stop
+            # them like this. Slurm will throw an error, but
+            # will actually kill steps correctly.
```

Comment:
Just curious: from my (admittedly limited) testing on horizon, I was never able to get `scancel` to return a non-zero exit code. I checked out the docs for `scancel` and noticed [this](https://slurm.schedmd.com/scancel.html#SECTION_DESCRIPTION) in its description.

Is there a chance that this was just erroring when the job was in a PENDING state? If so, and potentially out of scope, is there anyway we could check the "last" state of the job step previous to the `scancel` and use that information to decide if we should display the warning?

Reply:
Here is the reproducer, can you get an allocation with `salloc -N 3 : -N 1`, and then try it and let me know if it gives you the same error I get?

```python
from os import getenv
from smartsim import Experiment
from smartsim.error import SmartSimError
from smartsim.settings.slurmSettings import SrunSettings


exp = Experiment("hetjob", launcher="slurm")


db = exp.create_database(db_nodes=3, single_cmd=False)

het_size = getenv("SLURM_HET_SIZE")
if het_size:
    db.set_run_arg('het-group', '0')

model_rs: SrunSettings = exp.create_run_settings ('sleep', ['1'])
model_rs.set_het_group(int(het_size)-1)

model = exp.create_model("sleep", model_rs)

exp.generate(db, model, overwrite=True)

try:
    exp.start(db, model)
except SmartSimError as e:
    raise e from None
finally:
    exp.stop(db)
```

Reply:
Ahhh that is causing the warning to throw, and the jobs are indeed canceled. That's so weird :/

All well, if it works it works! We can work on mitigating the warning at a late date!!